internal/cli/test.go

```diff
@@ -99,56 +92,29 @@ func testLoginCmd(cli *cli) *cobra.Command {
-  auth0 test login <client-id> --connection <connection> --audience <audience> --domain <domain> --scopes <scope1,scope2>
-  auth0 test login <client-id> -c <connection> -a <audience> -d <domain> -s <scope1,scope2> --force
-  auth0 test login <client-id> -c <connection> -a <audience> -d <domain> -s <scope1,scope2> --json
-  auth0 test login <client-id> -c <connection> -a <audience> -d <domain> -s <scope1,scope2> --force --json`,
+  auth0 test login <client-id> --connection-name <connection-name>
+  auth0 test login <client-id> --connection-name <connection-name> --audience <api-identifier>
+  auth0 test login <client-id> --connection-name <connection-name> --audience <api-identifier> --domain <domain>
+  auth0 test login <client-id> --connection-name <connection-name> --audience <api-identifier> --domain <domain> --scopes <scope1,scope2>
+  auth0 test login <client-id> -c <connection-name> -a <api-identifier> -d <domain> -s <scope1,scope2> --force
+  auth0 test login <client-id> -c <connection-name> -a <api-identifier> -d <domain> -s <scope1,scope2> --json
+  auth0 test login <client-id> -c <connection-name> -a <api-identifier> -d <domain> -s <scope1,scope2> --force --json`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			const commandKey = "test_login"
-			var userInfo *authutil.UserInfo
-			isTempClient := false
-
-			tenant, err := cli.getTenant()
+			client, err := selectClientToUseForTestsAndValidateExistence(cli, cmd, args, &inputs)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to select client to use for tests: %w", err)
 			}
 
-			if len(args) == 0 {
-				err := testClientIDArg.Pick(cmd, &inputs.ClientID, cli.appPickerOptions)
-				if err != nil {
-					if err != errNoApps {
-						return err
-					}
-					cli.renderer.Infof("No applications to select from, we will create a default test application " +
-						"for you and remove it once the test is complete.")
-					client := &management.Client{
-						Name:             auth0.String(cliLoginTestingClientName),
-						Description:      auth0.String(cliLoginTestingClientDescription),
-						Callbacks:        &[]string{cliLoginTestingCallbackURL},
-						InitiateLoginURI: auth0.String(cliLoginTestingInitiateLoginURI),
-					}
-					if err := cli.api.Client.Create(client); err != nil {
-						return fmt.Errorf("Unable to create an app for testing the login box: %w", err)
-					}
-					inputs.ClientID = client.GetClientID()
-					isTempClient = true
-					cli.renderer.Infof("Default test application successfully created\n")
-				}
-			} else {
-				inputs.ClientID = args[0]
-			}
-
-			defer cleanupTempApplication(isTempClient, cli, inputs.ClientID)
-
-			client, err := cli.api.Client.Read(inputs.ClientID)
-			if err != nil {
-				return fmt.Errorf("Unable to find client %s; if you specified a client, please verify it exists, otherwise re-run the command", inputs.ClientID)
+			if client.GetAppType() == appTypeNonInteractive {
```

Comment:
We shouldn't try to run a login for M2M app types but use the auth0 test token command instead for those.